workitem/simple_type.go

```diff
@@ -176,6 +177,110 @@ func (t SimpleType) ConvertToModel(value interface{}) (interface{}, error) {
 	}
 }
 
+// ConvertToStringSlice implements the FieldType interface
+func (t SimpleType) ConvertToStringSlice(value interface{}) ([]string, error) {
+	if value == nil {
+		// if a value is nil, we return empty string.
+		return []string{""}, nil
+	}
+	valueType := reflect.TypeOf(value)
+	switch t.GetKind() {
+	case KindString, KindUser, KindIteration, KindArea, KindLabel, KindBoardColumn:
+		if valueType.Kind() != reflect.String {
+			return nil, errs.Errorf("value %v (%[1]T) should be %s, but is %s", value, "string", valueType.Name())
+		}
+		return []string{value.(string)}, nil
+	case KindURL:
+		if valueType.Kind() == reflect.String && govalidator.IsURL(value.(string)) {
+			return []string{value.(string)}, nil
+		}
+		return nil, errs.Errorf("value %v (%[1]T) should be %s, but is %q", value, "URL", valueType.Name())
+	case KindFloat:
+		if valueType.Kind() != reflect.Float64 {
+			return nil, errs.Errorf("value %v (%[1]T) should be %s, but is %q", value, "float64", valueType.Name())
+		}
+		fval, ok := value.(float64)
+		if !ok {
+			return nil, errs.Errorf("failed to cast value %+v (%[1]T) to float64", value)
+		}
+		return []string{strconv.FormatFloat(fval, 'f', 6, 64)}, nil
+	case KindInteger:
+		// NOTE(kwk): This will change soon to be more consistent.
+		switch valueType.Kind() {
+		case reflect.Int:
+			ival, ok := value.(int)
+			if !ok {
+				return nil, errs.Errorf("failed to cast value %+v (%[1]T) to int", value)
+			}
+			return []string{strconv.Itoa(ival)}, nil
+		case reflect.Int64:
+			ival, ok := value.(int64)
+			if !ok {
+				return nil, errs.Errorf("failed to cast value %+v (%[1]T) to int64", value)
+			}
+			return []string{strconv.FormatInt((ival), 10)}, nil
+		case reflect.Float64:
+			fval, ok := value.(float64)
+			if !ok {
+				return nil, errs.Errorf("failed to cast value %+v (%[1]T) to float64", value)
+			}
+			if fval != math.Trunc(fval) {
+				return nil, errs.Errorf("float64 value %+v (%[1]T) has digits after the decimal point and therefore cannot be represented by an integer", value)
+			}
+			return []string{strconv.FormatFloat(fval, 'f', 0, 64)}, nil
+		default:
+			return nil, errs.Errorf("value %v (%[1]T) should be %s, but is %s ", value, "int or float", valueType.Name())
+		}
+	case KindInstant:
+		if valueType.Kind() != timeType.Kind() {
+			return nil, errs.Errorf("value %v (%[1]T) should be %s, but is %s", value, "time.Time", valueType.Name())
+		}
+		return []string{(value.(time.Time)).Format(time.RFC3339)}, nil
```

Comment:
I'm sorry to reject this, but conversions without checks will panic which is unacceptable

```suggestion
		theTime, ok := value.(time.Time)
		if !ok {
				return nil, errs.Errorf(`failed to convert value to "time.Time": %+v`, value)
		}
		return []string{theTime.Format(time.RFC3339)}, nil
```

Reply:
@michaelkleinhenz have you addressed this meanwhile?